sympy/core/tests/test_function.py

```diff
@@ -639,7 +639,6 @@ def fdiff(self, idx):
             raise ArgumentIndexError
     mf = myfunc(x)
     assert mf.diff(x) == Derivative(mf, x)
-    raises(TypeError, lambda: myfunc(x, x))
```

Comment:
This **is** the regression in this PR. With the new code, custom classes that redefine `nargs` no longer get args length checking. We should not remove this test, but rather fix the code so that it continues to pass.